### PR TITLE
[Network] `az network private-endpoint-connection`: Add provider `Microsoft.DBforPostgreSQL/flexibleServers`

### PR DESCRIPTION
**Pull request by hakashya**
**Related command**
`az network private-endpoint create`
`az network private-endpoint-connection show/approve/list/reject`

**Description**
This PR onboards the Microsoft.DBforPostgreSQL/flexibleServers RP to use the `az network private-endpoint*` commands. This allows Azure customers to manage private endpoints and private endpoint connections to the flexible server resources using the `az network` module.

**Testing Guide**
Added integration tests and can be tested by running `azdev test NetworkPrivateLinkPostgreSQLFlexibleServerScenarioTest`.
Since the feature is in preview, the subscriptions making use of private endpoint for flexible servers must be whitelisted internally by the RP.

**History Notes**
 N/A

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Network

File: src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py [hunk 1]
```diff
@@ -81,7 +81,7 @@ def register_providers():
     _register_one_provider("Microsoft.MachineLearningServices/registries", '2022-10-01-preview', True)
     _register_one_provider('Microsoft.DBforMySQL/flexibleServers', '2022-09-30-privatepreview', False)
     _register_one_provider("Microsoft.DocumentDB/mongoClusters", '2023-03-01-preview', True)
-
+    _register_one_provider('Microsoft.DBforPostgreSQL/flexibleServers', '2023-06-01-preview', False)
```
Comment (paulmey) [inline src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py:85]: You should leave 2 empty lines here. It is custom to have 2 empty lines between function definitions in python code.

Comment (paulmey) [reply src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py:85]: this should also be fixed in the test source file.

**Review by paulmey — COMMENTED** [review]
Just formatting comments...

File: src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py (hunk 1 above)
Comment (hakashya) [reply src/azure-cli/azure/cli/command_modules/network/private_link_resource_and_endpoint_connections/custom.py:85]: Added 2 blank lines between each function definitions.

**Review by paulmey — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**